src/Tinfoil/Data/KDF.hs

```diff
@@ -64,7 +64,7 @@ data Verification = Verification !Verified !NeedsRehash
 --  processors (GPUs, mining ASICs, et cetera).
 data KDF = KDF
   { genHash        :: (Credential -> IO CredentialHash)
-  , hashCredential :: (Credential -> CredentialHash -> IO Verified)
-  , verifyNoHash   :: IO Verified
+  , hashCredential :: (CredentialHash -> Credential -> IO (Maybe Verified))
+  , verifyNoHash   :: (Credential -> IO (Maybe Verified))
```

Comment:
Ah, I Just mean:

```
data Verified = Verified | NotVerified | CanNotVerify
```

vs

```
data Verified = Verified | NotVerified -- with Maybe Verified at call site.
```
I don't really care that much, just thought it would look neater if both hashCredential and verifyNoHash just returned `IO Verified`.